cmd/snapshot/main.go

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2018 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"runtime"
+
+	"github.com/gardener/etcd-backup-restore/cmd/snapshot/app"
+)
+
+var onlyOneSignalHandler = make(chan struct{})
+
+func main() {
+	if len(os.Getenv("GOMAXPROCS")) == 0 {
+		runtime.GOMAXPROCS(runtime.NumCPU())
+	}
+
+	stopCh := setupSignalHandler()
+	command := app.NewCommandStartSnapshotter(stopCh)
+	if err := command.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+// SetupSignalHandler registered for SIGTERM and SIGINT. A stop channel is returned
```

Comment:
Why do we need 2 traps? One trap to enough to close, perform housekeeping tasks and exit gracefully.

Reply:
This is standard approach followed in majority projects. In case somebody don't want to wait till cleanup work.